mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3373,6 +3378,51 @@ static ModuleOp generateKernel(MLIRContext *context, GenParams &genParams,
   return module;
 }
 
+static ModuleOp populateCloneHarnessLogic(ModuleOp &module) {
+  if (arch.getValue().empty()) {
+    llvm::errs() << "--arch is not set\n";
+    exit(1);
+  }
+  func::FuncOp originalFunc = module.lookupSymbol<func::FuncOp>(testFuncName);
+  assert(originalFunc && "does -fut point to the wrong function?");
```

Comment:
Why an assert here?

Reply:
I copied from the clone verifier code to ensure we got a right function.